### PR TITLE
876-ittms-refactor-environment-variables-configuration

### PR DESCRIPTION
**Pull request by johnake**
## Context

To reduced the code maintenance overhead, we need to refactor some environment variables

## Changes proposed in this pull request

Create yml files per environment for the env variables: review_app_env.yml(in this case) and populate review_app_env.yml with values from [.env](https://github.com/DFE-Digital/itt-mentor-services/blob/main/.env.example)

## Guidance to review

Ensure single env variables are removed from variables.tf
review_app_env.yml has been created with the correct value and syntax
A mechanism to pass the env variables has been added to the application configuration

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated or [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by johnake** [comment]
[ITTMS: Refactor environment variables configuration](https://trello.com/c/81LvMRgU/876-ittms-refactor-environment-variables-configuration)

File: terraform/application/config/review_app_env.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+GOVUK_NOTIFY_API_KEY:
```
Comment (saliceti) [inline terraform/application/config/review_app_env.yml:2]: This variable is a secret and will come from keyvault

File: terraform/application/variables.tf [hunk 2]
```diff
@@ -38,17 +38,15 @@ variable "enable_monitoring" {
   default     = false
   description = "Enable monitoring and alerting"
 }
-variable "sign_in_method" {
-  default     = "persona"
-  description = "sign in method for the app"
-}
+
 variable "claims_host" {
```
Comment (saliceti) [inline terraform/application/variables.tf:42]: claims_host and placements_host can be removed from here as they are in the yaml file

File: terraform/application/application.tf [hunk 3]
```diff
@@ -11,13 +11,14 @@ module "application_configuration" {
...
   is_rails_application = true
 
-  config_variables = {
+  config_variables = merge(
+    var.app_env_values,
+    {
     ENVIRONMENT_NAME = var.environment
     PGSSLMODE        = local.postgres_ssl_mode
-    SIGN_IN_METHOD   = var.sign_in_method
     CLAIMS_HOST      = local.claims_host
```
Comment (saliceti) [inline terraform/application/application.tf:19]: CLAIMS_HOST and PLACEMENTS_HOST can be removed from here as they are in the yaml file

File: terraform/application/config/review.tfvars.json [hunk 4]
```diff
@@ -1,4 +1,5 @@
 {
+  "app_environment": "review",
```
Comment (saliceti) [inline terraform/application/config/review.tfvars.json:2]: The "config" variable already has this value: https://github.com/DFE-Digital/itt-mentor-services/blob/abf57b2ff9e1af7dd11a7b7e876f22f820da660f/global_config/review.sh#L1
It should be passed to terraform like config_short: https://github.com/DFE-Digital/itt-mentor-services/blob/abf57b2ff9e1af7dd11a7b7e876f22f820da660f/Makefile#L47

File: terraform/application/variables.tf [hunk 5]
```diff
@@ -10,6 +10,9 @@ variable "environment" {
 variable "azure_resource_prefix" {
   description = "Standard resource prefix. Usually s189t01 (test) or s189p01 (production)"
 }
+variable "config" {
+  description = "The application environment - should match the tfvars file name e.g. review, production..."
```
Comment (saliceti) [inline terraform/application/variables.tf:14]: The application ~~environment~~ configuration - 

(environment is the unique name of the deployed environment, like review-1234)

**Comment by saliceti** [comment]
> Review app track and pay deployed to https://track-and-pay-124.test.teacherservices.cloud Review app school placements deployed to https://manage-school-placements-124.test.teacherservices.cloud

The above URLs are not working anymore

File: terraform/application/config/review_app_env.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+---
+SIGN_IN_METHOD: persona
+PLACEMENTS_HOST: placements.localhost
+CLAIMS_HOST: claims.localhost
+GIAS_CSV_BASE_URL: https://ea-edubase-api-prod.azurewebsites.net/edubase/downloads/public
+PUBLISH_BASE_URL: https://www.publish-teacher-training-courses.service.gov.uk
```
Comment (RMcVelia) [inline terraform/application/config/review_app_env.yml:6]: qa.api.publish-teacher-training-courses.service.gov.uk

**Review by VenkataChalla554 — APPROVED** [review]

**Pull request merged**